app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -372,10 +372,11 @@ protected function validateSubmittedOrder($order, $quote) {
      * @param string $data     an object to be encoded to JSON as the value passed to the endpoint
      * @param string $object   defines part of endpoint url which is normally/always??? set to merchant
      * @param string $type     Defines the endpoint type (i.e. order|transactions|sign) that is used as part of the url
+     * @param null $storeId
      * @throws  Mage_Core_Exception  thrown if an error is detected in a response
      * @return mixed           Object derived from Json got as a response
      */
-    public function transmit($command, $data, $object='merchant', $type='transactions')
+    public function transmit($command, $data, $object='merchant', $type='transactions', $storeId = null)
     {
         $url = $this->getApiUrl() . 'v1/';
```

Comment:
do you have to pass storeId here?

Reply:
This is the cleanest way to apply the store lookup. If it's not done this way it will require some ugly code to pull it.